File: Leetcode/1486.py
# 1486. 数组异或操作 模拟
# link：https://leetcode-cn.com/problems/xor-operation-in-an-array/

class Solution:
    def xorOperation(self, n: int, start: int) -> int:
        result: int = 0
        cur: int = start
        for _ in range(n):
            result ^= cur
            cur += 2
        return result

File: Leetcode/1486.cpp
/**
 * 1486. 数组异或操作 模拟
 * link：https://leetcode-cn.com/problems/xor-operation-in-an-array/
 */

class Solution
{
public:
    int xorOperation(int n, int start)
    {
        int result = 0;
        int cur = start;
        while (n-- > 0)
        {
            result ^= cur;
            cur += 2;
        }
        return result;
    }
};